### PR TITLE
RHINENG-11246: Fixed inefficient SQL issue count sub-select and added db index

### PR DESCRIPTION
**Pull request by rexwhite**
Improving SQL query for fetching remediation plan by optimizing. issue count sub-select and adding a table index.

### CODE REVIEW

File: db/migrations/20240820140920-add-index-on-issue-resolution.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+module.exports = {
+  async up (q) {
+    q.addIndex('remediation_issue_systems', ['resolved', 'remediation_issue_id']);
```
Comment (vkrizan) [inline db/migrations/20240820140920-add-index-on-issue-resolution.js:5]: Any reason to have the `resolved` first? `remediation_issue_id` has higher cardinality.

Comment (rexwhite) [reply db/migrations/20240820140920-add-index-on-issue-resolution.js:5]: Nope - that's just the way I typed it in my cold medicine induced haze...

Comment (rexwhite) [reply db/migrations/20240820140920-add-index-on-issue-resolution.js]: Fixed.

**Review by vkrizan — APPROVED** [review]

**Comment by sthirugn** [comment]
@rexwhite++ @vkrizan++ thank you for the collaboration here

**Pull request merged**